### PR TITLE
docs: add documentation for authorization

### PR DESCRIPTION
**Pull request by deepakrkris**
issue: https://github.com/strongloop/loopback-next/issues/3695

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -110,6 +110,41 @@ possible by the use of the `UserService` service provided by
 You can see the details in
 [`packages/shopping/src/controllers/user.controller.ts`](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/controllers/user.controller.ts).
 
+### Authorization
+
+For an authorization demo in this example, all requests to the endpoint
```
Comment (dhmlau) [inline README.md]: I personally find it a bit confusing to consider it as a demo, below is my suggested change, feel free to take it as appropriate. 
From:
`For an authorization demo in this example` 
to:
`To see authorization in action in this example`

File: README.md [hunk 2]
```diff
@@ -110,6 +110,41 @@ possible by the use of the `UserService` service provided by
...
+user access.
+
+Endpoint authorization is done using `@loopback/authorization` package and the
+provisions in it are implemented using
+[Casbin](https://github.com/casbin/casbin) configurations.
+
+The following is a typical scenario for an authorization demo:
+
+1. An existing user credentials is used to login using the endpoint
+   `/users/login`. The JWT token returned has the user name (lets say the user's
```
Comment (dhmlau) [inline README.md]: `lets say` -> `let's say`

File: README.md [hunk 3]
```diff
@@ -110,6 +110,41 @@ possible by the use of the `UserService` service provided by
...
+
+For an authorization demo in this example, all requests to the endpoint
+`/users/{userId}/orders` (i.e, `UserOrderController`) are validated for specific
+user access.
+
+Endpoint authorization is done using `@loopback/authorization` package and the
+provisions in it are implemented using
+[Casbin](https://github.com/casbin/casbin) configurations.
+
+The following is a typical scenario for an authorization demo:
```
Comment (dhmlau) [inline README.md]: maybe: `Below is a scenario illustrating how authorization works in this example`?

File: README.md [hunk 4]
```diff
@@ -110,6 +110,41 @@ possible by the use of the `UserService` service provided by
...
+
+1. An existing user credentials is used to login using the endpoint
+   `/users/login`. The JWT token returned has the user name (lets say the user's
+   name is `bob`) and the user's id (lets say bob's id is `123`).
```
Comment (dhmlau) [inline README.md]: same here for `lets say`

File: README.md (hunk 2 above)
Comment (dhmlau) [reply README.md]: I'm thinking maybe we can reword as follows:
```
Bob logs into the shopping app using his existing user credential via endpoint `/users/login`.  The JWT token.....
```

If we're going to this route, the following bullet would be in the form of "Bob wants to get his orders...." etc.

File: README.md (hunk 1 above)
Comment (deepakrkris) [reply README.md]: agreed

File: README.md (hunk 2 above)
Comment (deepakrkris) [reply README.md]: that is correct, I will reword with a user based example in mind

File: README.md (hunk 4 above)
Comment (deepakrkris) [reply README.md]: I am going to reword this entirely with user bob's actions

File: README.md (hunk 3 above)
Comment (deepakrkris) [reply README.md]: agreed

**Pull request merged**